opendota.py

```diff
@@ -8,8 +8,133 @@
 
 # get/find matches
 
+def generate_query(*heroes):
+    """generates an SQL query where hero1+hero2 is vs hero3+hero4"""
+    #  won't support 2v3's or 1v3's in the case of trilanes
+
+    def select(*heroes): 
+        """generates select statement for certain num of heroes"""
+        select = f"SELECT"
+
+
+        i = 0 #leave me be sir 
+        for data in heroes:
+            for hero in data: 
+                if i == 0: 
+                    select = select + f" player_matches.hero_id AS hero1, \n"
+                else: 
+                    select = select + f"player_matches{i}.hero_id AS hero{i+1}, \n"
+                i = i+1
+            i = 0
+            for hero in data: 
+                if i == 0: 
+                    select = select + f"player_matches.player_slot AS ps1, \n"
+                else: 
+                    select = select + f"player_matches{i}.player_slot AS ps{i+1}, \n"
+                i = i+1
+
+        select = select + f"matches.match_id,\n"
+        select = select + f"leagues.name leaguename\n"
+        return select
+
+    def join(*heroes): 
+        """generates join statements for certain num of heroes"""
+        join = f"FROM player_matches \n"
+        i = 0
+        for data in heroes: 
+            for hero in data:
+                if i == 0: 
+                    i = i+1
+                    continue
+                else:
+                    join = join + f"JOIN player_matches AS player_matches{i}\n"
+                    join = join + f"ON player_matches.match_id = player_matches{i}.match_id\n"
+                    i = i+1
+        join = join + f"JOIN matches\n"
+        join = join + f"ON player_matches.match_id = matches.match_id\n"
+        join = join + f"JOIN leagues using(leagueid)\n"
+        return join
+
+    def where(*heroes): 
+        """generates where statement for certain num of heroes"""
+        i = 0
+        where = f"WHERE"
+        for data in heroes: 
+            for hero in data: 
+                if i == 0: 
+                    where = where + f" player_matches.hero_id = {get_hero_id(hero)}\n"
+                    i = i+1
+                else: 
+                    where = where + f"AND player_matches{i}.hero_id = {get_hero_id(hero)}\n"
+                    i = i+1
```

Comment:
only difference is `player_matches` vs `player_matches{i}`, just make them all `player_matches{i}`